pkg/controller/windowsmachine/windowsmachine_controller.go

```diff
@@ -264,60 +265,89 @@ func (r *ReconcileWindowsMachine) Reconcile(request reconcile.Request) (reconcil
 	provisionedPhase := "Provisioned"
 	// runningPhase is the status of the machine when it is in the `Running` state, indicating that it is configured into a node
 	runningPhase := "Running"
+	// deletingPhase is the status of the machine when it is in the `Deleting` state
+	deletingPhase := "Deleting"
 	if machine.Status.Phase == nil {
 		// Phase is nil and should be ignored by WMCO until phase is set
 		// TODO: Instead of requeuing ignore certain events: https://issues.redhat.com/browse/WINC-500
 		return reconcile.Result{}, fmt.Errorf("could not get the phase associated with machine %s", machine.Name)
-	} else if *machine.Status.Phase == runningPhase {
-		// Machine has been configured into a node, we need to ensure that the version annotation exists. If it doesn't
-		// the machine was not fully configured and needs to be configured properly.
-		if machine.Status.NodeRef == nil {
-			// NodeRef missing. Requeue and hope it is created. It never being created indicates an issue with the
-			// machine api operator
-			return reconcile.Result{}, fmt.Errorf("ready Windows machine %s missing NodeRef", machine.GetName())
+	} else if *machine.Status.Phase == runningPhase || *machine.Status.Phase == deletingPhase {
```

Comment:
What happens if the machine is fully deleted before we can reconcile it?

Reply:
https://github.com/openshift/windows-machine-config-operator/pull/142/files?file-filters%5B%5D=.go&file-filters%5B%5D=.txt&file-filters%5B%5D=No+extension#diff-f6cf75653deff7ed72be94f059acc292d15113ed7c9f9ce0fdc5e842fa71aa72R256-R259

Reply:
@sebsoto Did not see this scenario in testing. I am not sure about it